test/unit/collector/serverless.test.js

```diff
@@ -14,10 +14,14 @@ const nock = require('nock')
 const chai = require('chai')
 const expect = chai.expect
 const sinon = require('sinon')
+const fs = require('fs')
 const helper = require('../../lib/agent_helper')
 const API = require('../../../lib/collector/serverless')
 const serverfulAPI = require('../../../lib/collector/api')
 
+const pipePath = '/tmp/newrelic-telemetry'
+const shouldUsePipe = fs.existsSync(pipePath)
```

Comment:
Same as above. Introducing variable execution into a test seems dangerous. I'd prefer we isolate the different paths, if necessary to cover both, by controlling the state for the tests.